tests/unit/backend/wmg/api/test_expression_rollup.py

```diff
@@ -41,3 +44,119 @@ def test__expression_rollup_no_descendants_overlap(self):
+
+        cell_types_dot_plot_df = cell_types_cell_counts_df[:num_cell_types_with_expression] * num_tissues * num_genes
+        tissue_dot_plot_df = (
+            sum([[f"tissue_ontology_term_id_{i}"] * num_cell_types_with_expression for i in range(num_tissues)], [])
+            * num_genes
+        )
+        genes_dot_plot_df = sum(
+            [[f"gene_ontology_term_id_{i}"] * num_cell_types_with_expression * num_tissues for i in range(num_genes)],
+            [],
+        )
+
+        cell_counts_cell_type_agg = pd.DataFrame()
+        cell_counts_cell_type_agg["cell_type_ontology_term_id"] = cell_types_cell_counts_df
+        cell_counts_cell_type_agg["tissue_ontology_term_id"] = tissue_cell_counts_df
+        cell_counts_cell_type_agg["n_cells_cell_type"] = 1
+
+        dot_plot_matrix_df = pd.DataFrame()
+        dot_plot_matrix_df["cell_type_ontology_term_id"] = cell_types_dot_plot_df
+        dot_plot_matrix_df["tissue_ontology_term_id"] = tissue_dot_plot_df
+        dot_plot_matrix_df["gene_ontology_term_id"] = genes_dot_plot_df
+        for attr in expression_summary_logical_attrs:
+            dot_plot_matrix_df[attr.name] = 1
+        dot_plot_matrix_df["n_cells_cell_type"] = 1
+        dot_plot_matrix_df["n_cells_tissue"] = 100
+
+        expected_added_cell_types = (
+            [f"cell_type_ontology_term_id_{i}" for i in range(num_cell_types_with_expression, num_cell_types)]
+            * num_tissues
+            * num_genes
+        )
+        expected_added_tissues = (
+            sum(
+                [
+                    [f"tissue_ontology_term_id_{i}"] * (num_cell_types - num_cell_types_with_expression)
+                    for i in range(num_tissues)
+                ],
+                [],
+            )
+            * num_genes
+        )
+        expected_added_genes = sum(
+            [
+                [f"gene_ontology_term_id_{i}"] * (num_cell_types - num_cell_types_with_expression) * num_tissues
+                for i in range(num_genes)
+            ],
+            [],
+        )
+        expected_added_dataframe = pd.DataFrame()
+        expected_added_dataframe["cell_type_ontology_term_id"] = expected_added_cell_types
+        expected_added_dataframe["tissue_ontology_term_id"] = expected_added_tissues
+        expected_added_dataframe["gene_ontology_term_id"] = expected_added_genes
+        for attr in expression_summary_logical_attrs:
+            expected_added_dataframe[attr.name] = 0
+        expected_added_dataframe["n_cells_cell_type"] = 0
+        expected_added_dataframe["n_cells_tissue"] = 100
+
+        expected_dot_plot_matrix_df = pd.concat((dot_plot_matrix_df, expected_added_dataframe), axis=0)
+
+        # group by to get cell_counts_cell_type_agg into the right format (with multi-index)
+        cell_counts_cell_type_agg = cell_counts_cell_type_agg.groupby(
+            ["cell_type_ontology_term_id", "tissue_ontology_term_id"]
+        ).first()
```

Comment:
The purpose of `groupby(...).first()` here is to reform the pandas DataFrame into one where the group-by terms are in a multi index, because that's what the `rollup` function is expecting. Does that address your comment?

Reply:
I see. It is my lack of knowledge about `pandas`